doc/manual.rst

```diff
@@ -4390,6 +4390,38 @@ parameters of an outer factory proc:
   for f in foo():
     echo f
 
+Parameter Passing For Iterators In `for` Loops
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+  The following only applies to `inline` iterators at the moment.
+
+For iterators invoked in the context of `for`-loops, the arguments are
+evaluated before passing control to the iterator's body. The argument
+evaluation order is the same as everywhere else, that is, left to right.
+
+How exactly this is achieved for `inline` iterators is up to the compiler, but
+with the following restrictions / guarantees:
+- no full copy must be introduced when passing to an immutable parameter (a full
+  copy might still be needed in order to ensure the left-to-right evaluation
+  order)
+- except for `seq`s, `string`s, and `ref`s, if the argument is an lvalue
+  expression, the parameter must not be a full location, but only store the
+  handle
+- if the argument is an unamed location (e.g. the result of a call), a full
+  location *may* be used
```

Comment:
Just noticed this upon re-reading, but we don't have a definition for "full location" prior to this, don't think it needs to be sorted in this PR.

I take it to mean something that stores a direct value, as opposed to a handle (`ref`/`ptr`) or a forwarded expression.

Reply:
Yeah, I was a bit mindless with the term here. I think the more proper description would be "owning location". From a quick skim of the manual, that term also doesn't seem to be defined anywhere yet, but I think it's a bit more self-explanatory (and correct) than "full location", so I'm going to change "full" to "owning" here.

Reply:
I updated it.